app/controllers/users/articles_controller.rb

```diff
@@ -20,11 +20,31 @@ def authorize_resource!
     end
 
     def authorized_articles
-      authorized(Article.all)
+      authorized(Article.all.includes(:comments))
     end
 
     def article_params
       params.require(:article).permit(:title, :content).merge(user: current_user)
     end
+
+    def fetch_article
+      if params[:id].present?
+        Article.find(params[:id])
+      elsif current_company.present?
+        fetch_company_article
+      elsif params[:article].present?
+        Article.new(article_params)
+      else
+        Article.new
+      end
+    end
+
+    def fetch_company_article
+      if params[:article].present?
+        current_company.articles.build(article_params)
+      else
+        current_company.articles.build
+      end
+    end
```

Comment:
```suggestion
   def article_params
  params.require(:article).permit(:title, :content).merge(user: current_user, comapny: current_company)
end

def fetch_article
  if params[:id].present?
    Article.find(params[:id])
  elsif params[:article].present?
    Article.new(article_params)
  else
    Article.new(comapny: current_company)
  end
end
```

Reply:
If company is blank it will leave it blank. And I added user in cases then we have not id and article in params. As I understood it is needed here. If I`m wrong, sorry